lib/elixir/lib/module/locals_tracker.ex

```diff
@@ -213,15 +213,24 @@ defmodule Module.LocalsTracker do
 
   defp get_line(meta), do: Keyword.get(meta, :line)
 
+  defp get_position(meta), do: maybe_add_col([line: get_line(meta)], meta)
+
   defp build_meta(nil, _meta), do: []
 
   # We need to transform any file annotation in the function
   # definition into a keep annotation that is used by the
   # error handling system in order to respect line/file.
-  defp build_meta(line, meta) do
+  defp build_meta(position, meta) do
     case Keyword.get(meta, :file) do
-      {file, _} -> [keep: {file, line}]
-      _ -> [line: line]
+      {file, _} -> maybe_add_col([keep: {file, position[:line]}], meta)
+      _ -> maybe_add_col(position, meta)
+    end
+  end
+
+  defp maybe_add_col(pos, meta) do
+    case meta[:column] do
+      nil -> pos
+      col -> Keyword.put(pos, :column, col)
```

Comment:
```suggestion
      col -> [column: col] ++ pos
```

(we know there is no `:column`, so no need to traverse for `put`)